### PR TITLE
ci: fix lint and add husky hook

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Issue Description

Makes a number of improvements to Lint.

Adds indentation rules, no code changes.

Related issue: #1783

### Approach

Add pre-commit hook, fixes existing lint rules.

### TODOs before merging

- [x] Add tests
- [x] Add entry to changelog
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)

### CODE REVIEW

**Comment by dblythy** [comment]
I would also like to add prettier but wanted to keep this PR minimal

**Comment by dblythy** [comment]
Also related to #1789

File: scripts/generate.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ function padding(length) {
 
 function generateReact(name) {
   return (
-`/*
+    `/*
```
Comment (mtrezza) [inline scripts/generate.js:29]: That looks odds...

Comment (dblythy) [reply scripts/generate.js:29]: I'm assuming it's caused by `"indent": ["error", 2, { "SwitchCase": 1 }],`?

Comment (dblythy) [reply scripts/generate.js:29]: I'm assuming it's caused by `"indent": ["error", 2, { "SwitchCase": 1 }],`? I can remove the new rules and tackle them in a different PR, if you would prefer

Comment (mtrezza) [reply scripts/generate.js:29]: Are these rules the same as for parse server?

Comment (dblythy) [reply scripts/generate.js:29]: Yep!

Comment (mtrezza) [reply scripts/generate.js:29]: Then let's just ignore this for now I suggest

**Comment by mtrezza** [comment]
Should we merge this now or wait until we have a few more PRs merged?

**Comment by dblythy** [comment]
Your call! It's all auto-fixed anyway so it can wait

**Comment by mtrezza** [comment]
Then I'd say let's merge a few more larger PRs first

**Comment by mtrezza** [comment]
@dblythy I think after merging https://github.com/parse-community/parse-dashboard/pull/1657 this PR should be good to go, we have merged all other larger PRs since Sept 10. The other larger PRs (https://github.com/parse-community/parse-dashboard/pull/1677, https://github.com/parse-community/parse-dashboard/pull/1761) have become stale and it doesn't seem as if they'd be picked up anytime soon unfortunately.

I would still want us to consider whether we need enforcing pre-commit hooks, I personally am not in favor of these for reasons I have mentioned before.

**Pull request closed without merging**